src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw Error('Length can\'t be equal or less than 0');
+    }
+
+    if (a >= b + c
+      || b >= a + c
+      || c >= a + b
```

Comment:
... or you can sort sides and check if the biggest one is '>=' sum of another ones. But this solution is good!